addons/account/wizard/account_move_reversal.py

```diff
@@ -72,7 +72,7 @@ def default_get(self, fields):
         if any(move.state != "posted" for move in move_ids):
             raise UserError(_('You can only reverse posted moves.'))
         if 'company_id' in fields:
-            res['company_id'] = move_ids.company_id.root_id.id or self.env.company.id
+            res['company_id'] = move_ids.company_id.id or self.env.company.id
```

Comment:
What happens if you have moves from multiple companies/subsidiaries?

Reply:
Indeed, as you guessed it, it will throw a singleton error, do you think, throwing a UserError is a good idea?

Reply:
```suggestion
            res['company_id'] = (max(move_ids.company_id, key=lambda c: len(c.parent_ids)) or self.env.company).id
```

Reply:
Hello,
But this will cause the same issue of having a reversal move related to Company **A** in a journal related to Company **B** and will throw the same access error in the following line: https://github.com/odoo/odoo/blob/17.0/odoo/models.py#L4039

Reply:
It shouldn't because it complies with 
https://github.com/odoo/odoo/blob/3992e88e82aa18bde2ebcf8330fe92257f548e50/addons/account/models/account_account.py#L19
https://github.com/odoo/odoo/blob/3992e88e82aa18bde2ebcf8330fe92257f548e50/odoo/models.py#L175-L187